lib/client-assets.php

```diff
@@ -52,10 +53,8 @@ function gutenberg_url( $path ) {
  * @param bool             $in_footer Optional. Whether to enqueue the script before </body> instead of in the <head>.
  *                                    Default 'false'.
  */
-function gutenberg_override_script( $handle, $src, $deps = array(), $ver = false, $in_footer = false ) {
-	global $wp_scripts;
-
-	$script = $wp_scripts->query( $handle, 'registered' );
+function gutenberg_override_script( &$scripts, $handle, $src, $deps = array(), $ver = false, $in_footer = false ) {
+	$script = $scripts->query( $handle, 'registered' );
 	if ( $script ) {
```

Comment:
@mcsf - do you think, we still need this complex logic to override the script, or could we do the same trick as for CSS where we simply remove it and add again? We are operating on `wp_default_scripts` hook now, which happens just after those scripts get registered in the WordPress core, before they are processed. That's why we can't use wrapping function and need to operate on the `$scripts` object passed by reference.

Reply:
https://travis-ci.com/WordPress/gutenberg/jobs/243283375#L1081
```
1) Override_Script_Test::test_registers_new_script
Undefined index: group
/var/www/src/wp-content/plugins/gutenberg/phpunit/class-override-script-test.php:87
```

We will probably have to do it regardless.

Reply:
I figured out that we have to keep this logic. I don't fully understand why though. We need to update the test which registers a new script:
https://github.com/WordPress/gutenberg/pull/17576/commits/3a7aacc3c254831ee7bafe55e383cf7a9a7f9a3d

Reply:
I did some further debugging and I figured out that we don't want to lose inline scripts and the following is what would be the closest to what we need:

https://github.com/WordPress/gutenberg/pull/17576/commits/ec76faf2327302696a6ed7ab16d6e04d674d126c

Reply:
I had to bring back some more logic which handles script overrides to ensure that scripts are printed in the footer as intended:
https://github.com/WordPress/gutenberg/pull/17576/commits/812b93c4df4aeab5e964c5500f43af12b9bca817

Reply:
> do you think, we still need this complex logic to override the script, or could we do the same trick as for CSS where we simply remove it and add again?

I'm not sure. In the past I only worked on the remove-and-readd approach (https://github.com/WordPress/gutenberg/commit/fd64343eca66f3aff0b3a9298f2ebd4f331d6440?w=1#diff-620130744b6b73b822a3df609671d930R109-R112). This more complex logic is mostly Andrew's.

From a look at both those changes and the source for dependencies, it's clear that if we want to preserve attached inline scripts there's a lot of legwork, so I don't see how much we can simplify here.